RIOT-OS/RIOT
# pkg/nrf_modem: add initial modem support with gnss service

### Contribution description
This PR adds initial support for nRF9160 modem library as a RIOT pkg.
The modem library (and the modem firmware) allow to use NB-IoT, LTE-M and GNSS features for nRF9160 MCUs.
In the current state, only the GNSS service is added. (NB-IoT and LTE-M can be added in a followup PR).

MCU and modem can communicate through the device's RAM and is managed by the Inter-Processor Communication peripheral (IPC).

Unfortunately, the modem library relies on Nordic drivers SDK for handling IPC stuff, thus I add to "emulate" somehow some functions of the Nordic SDK to make the modem library happy.

I think some discussions are needed for specific like the malloc management because some malloc() are made from ISR context by the modem library which is troublesome. 

### Testing procedure
Run `tests/pkg_nrf_modem` on `nRF9160DK`, connect your terminal and go for a quick walk outside.
Please make sure the nRF modem firmware is flashed **before** testing the application.
You can flash it through nrfconnect application or you can use `make nrf_modem/flash_firmware` which will use `nrfjprog` tool.
(This PR also adds nrfjprog as PROGRAMMER).

The application will initialize the GNSS service  and will wait for a GPS fix event. Once GPS fix is made, NMEA frame will be displayed on the shell with your current location.

### Issues/PRs references
None.
